vertx-mysql-client/src/main/java/io/vertx/mysqlclient/impl/codec/MySQLEncoder.java

```diff
@@ -49,6 +49,14 @@ void write(CommandBase<?> cmd) {
     codec.completionHandler = resp -> {
       CommandCodec c = inflight.poll();
       resp.cmd = (CommandBase) c.cmd;
+      /*
+       * a bit hacky but we need this message delivered to the socket messageHandler
+       */
+      if (resp.cause() != null && resp.cause().getMessage().equals("SSL handshake failed")) {
```

Comment:
this does seem a bit hacky, where does the error message come from? Is there any chance it could change or be a localized (non-english) message?

Reply:
The error is from https://github.com/eclipse-vertx/vertx-sql-client/blob/6d38d4b9f442642ecc6b6de425e23d012cd7270b/vertx-mysql-client/src/main/java/io/vertx/mysqlclient/impl/InitiateSslHandler.java#L56

Because in Vert.x 3.8 the `NetSocket#upgradeToSsl` method does not have an async result to represent if it fails, so we let the failure propagated in the channel pipeline and gets the connection closed, then we can check the activity of channel to know if the SSL handshake is successful or not. We only need this trick in `3.8` branch which is different what I do in my another PR for `master` branch(i.e. 4.0 version).